From 5a323cf99ca449e4d47014ce10f9da39f902bf29 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 21 Jul 2011 11:18:58 +0800
Subject: [PATCH 316/691] ENGR00152843 mx51/3: align the android default env
 with document

Use simple enviroment to implement the default boot command.
The original one is too complex, and not readable.
For MX51BBG, only SD card boot env is supportd by default.
For MX53SMD, only eMMC boot env is supportd by default.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 include/configs/mx51_bbg_android.h |   42 ++++++++---------------------------
 include/configs/mx53_smd_android.h |   40 +++++++++------------------------
 2 files changed, 21 insertions(+), 61 deletions(-)

diff --git a/include/configs/mx51_bbg_android.h b/include/configs/mx51_bbg_android.h
index 75d458c..16e92b8 100644
--- a/include/configs/mx51_bbg_android.h
+++ b/include/configs/mx51_bbg_android.h
@@ -159,42 +159,20 @@
 #define CONFIG_PRIME	"FEC0"
 
 #define CONFIG_LOADADDR		0x90800000	/* loadaddr env var */
-#define CONFIG_RD_LOADADDR	(CONFIG_LOADADDR + 0x400000)
+#define CONFIG_RD_LOADADDR	(CONFIG_LOADADDR + 0x500000)
 
 #define	CONFIG_EXTRA_ENV_SETTINGS					\
 		"netdev=eth0\0"						\
 		"ethprime=FEC0\0"					\
-		"uboot_addr=0xa0000000\0"				\
-		"uboot=u-boot.bin\0"			\
-		"kernel=uImage\0"				\
-		"rd_loadaddr=0x90D00000\0"	\
-		"nfsroot=/opt/eldk/arm\0"				\
-		"bootargs_base=setenv bootargs console=ttymxc0,115200 gpu_nommu,gpu_memory=32M\0"\
-		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs " \
-			"ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"\
-		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
-			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
-		"prg_uboot=tftpboot ${loadaddr} ${uboot}; "		\
-			"protect off ${uboot_addr} 0xa003ffff; "	\
-			"erase ${uboot_addr} 0xa003ffff; "		\
-			"cp.b ${loadaddr} ${uboot_addr} ${filesize}; "	\
-			"setenv filesize; saveenv\0"			\
-		"bootcmd=run bootcmd_SD \0"				\
-		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 2000;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x3000 0x300;"		\
-		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
-		"bootargs_android=setenv bootargs ${bootargs}  "	\
-		     "androidboot.console=ttymxc0 init=/init "		\
-		     "di1_primary calibration\0"			\
-		"bootcmd_android_recovery=run bootargs_base"		\
-		     " bootargs_android_recovery;"			\
-		     "mmc read 0 ${loadaddr} 0x800 0x2000;bootm\0"	\
-		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
-		     " init=/init root=/dev/mmcblk0p4 rootfs=ext4"	\
-		     " di1_primary \0"					\
-
-
+		"ethaddr=00:04:9f:00:ea:d3\0"		\
+		"bootfile=uImage\0"	\
+		"loadaddr=0x90800000\0"				\
+		"rd_loadaddr=0x90D00000\0"		\
+		"bootargs=console=ttymxc0 init=/init androidboot.console=ttymxc0 " \
+			"di1_primary gpu_nommu gpu_memory=32M calibration\0" \
+		"bootcmd_SD=mmc read 0 ${loadaddr} 0x800 0x2000;" \
+			"mmc read 0 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 
 /*Support LAN9217*/
diff --git a/include/configs/mx53_smd_android.h b/include/configs/mx53_smd_android.h
index ddc6bd2..5013cfb 100644
--- a/include/configs/mx53_smd_android.h
+++ b/include/configs/mx53_smd_android.h
@@ -152,39 +152,21 @@
 #define CONFIG_PRIME	"FEC0"
 
 #define CONFIG_LOADADDR		0x70800000	/* loadaddr env var */
-#define CONFIG_RD_LOADADDR	(CONFIG_LOADADDR + 0x400000)
+#define CONFIG_RD_LOADADDR	(CONFIG_LOADADDR + 0x500000)
 
 #define	CONFIG_EXTRA_ENV_SETTINGS					\
 		"netdev=eth0\0"						\
 		"ethprime=FEC0\0"					\
-		"uboot=u-boot.bin\0"					\
-		"kernel=uImage\0"					\
-		"loadaddr=0x70800000\0"					\
-		"rd_loadaddr=0x70D00000\0"				\
-		"nfsroot=/opt/eldk/arm\0"				\
-		"bootargs_base=setenv bootargs console=ttymxc0,115200\0"\
-		"bootargs_nfs=setenv bootargs ${bootargs} root=/dev/nfs " \
-		    "ip=dhcp nfsroot=${serverip}:${nfsroot},v3,tcp\0"	 \
-		"bootargs_android=setenv bootargs ${bootargs} "		\
-		     "androidboot.console=ttymxc0 init=/init di1_primary "\
-		     "video=mxcdi1fb:RGB666,XGA hdmi ldb=di1 gpu_nommu,gpu_memory=64M\0"   \
-		"bootcmd_SD=run bootargs_base bootargs_android;"	\
-		     "mmc read 0 ${loadaddr} 0x800 2000;"		\
-		     "mmc read 0 ${rd_loadaddr} 0x3000 0x300;"		\
-		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
-		"bootcmd_eMMC=run bootargs_base bootargs_android;"	  \
-		     "mmc read 1 ${loadaddr} 0x800 2000;"		\
-		     "mmc read 1 ${rd_loadaddr} 0x3000 0x300;"		\
-		     "bootm ${loadaddr} ${rd_loadaddr}\0"		\
-		"bootcmd_net=run bootargs_base bootargs_nfs; "		\
-			"tftpboot ${loadaddr} ${kernel}; bootm\0"	\
-		"bootcmd_android_recovery=run bootargs_base"		\
-		     " bootargs_android_recovery;"			\
-		     "mmc read 1 ${loadaddr} 0x800 0x2000;bootm\0"	\
-		"bootargs_android_recovery=setenv bootargs ${bootargs}" \
-		     "init=/init root=/dev/mmcblk0p4 rootfs=ext4 "	\
-		     "di1_primary video=mxcdi1fb:RGB666,XGA ldb=di1\0"	  \
-		"bootcmd=run bootcmd_eMMC\0"
+		"ethaddr=00:04:9f:00:ea:d3\0"		\
+		"bootfile=uImage\0"	\
+		"loadaddr=0x70800000\0"				\
+		"rd_loadaddr=0x70D00000\0"		\
+		"bootargs=console=ttymxc0 init=/init " \
+			"androidboot.console=ttymxc0 video=mxcdi1fb:RGB666,XGA " \
+			"ldb=di1 di1_primary gpu_nommu gpu_memory=64M\0" \
+		"bootcmd_SD=mmc read 1 ${loadaddr} 0x800 0x2000;" \
+			"mmc read 1 ${rd_loadaddr} 0x3000 0x300\0" \
+		"bootcmd=run bootcmd_SD; bootm ${loadaddr} ${rd_loadaddr}\0" \
 
 #define CONFIG_ARP_TIMEOUT	200UL
 
-- 
1.7.7.4

